core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCacheImpl.java

```diff
@@ -692,64 +640,136 @@ private void requestTabletHosting(ClientContext context,
     }
   }
 
-  private void lookupTablet(ClientContext context, Text row, boolean retry,
-      LockCheckerSession lcSession) throws AccumuloException, AccumuloSecurityException,
-      TableNotFoundException, InvalidTabletHostingRequestException {
+  private static class ReferenceCountedLock {
```

Comment:
Can you add a javadoc to explain how this is supposed to be used correctly?

Reply:
Moved the functionality into its own class named LockMap in 7a702e4 where it can be documented and tested independently.  In a follow on PR could made some code in tablet server that is doing something similar use this new class.